src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameGapCheck.java

```diff
@@ -113,16 +123,23 @@ public boolean validCheckForObject(final AtlasObject object)
                 outEdges);
         if (matchingInAndOutEdgeNames.isEmpty())
         {
-            // There is no pair of inedge and out edge with same name.
+            // There is no pair of in edge and out edge with same name.
             return Optional.empty();
         }
-
+        final String nameSuggestion = matchingInAndOutEdgeNames.iterator().next();
```

Comment:
Related to [this example](https://github.com/osmlab/atlas-checks/pull/475#pullrequestreview-572247523), I think we should be careful assigning an arbitrary name if there is more than one in `matchingInAndOutEdgeNames`. This can happen if pairs of in/out edges are within 9 degrees of heading. I see there is only 1 FP left so ill leave this up to you, but to fix such cases, we might be able to return an empty optional if `matchingInAndOutEdgeNames` is empty OR `edge` has a valence of > 1 at its start and end nodes

Reply:
Do we want to go ahead and merge this, or should we address this before the release?